### PR TITLE
Airspeed selector: fix condition for messaging

### PR DESCRIPTION
**Pull request by fury1895**
**Describe problem solved by this pull request**
Current behavior: If airspeed has been valid and gets invalid prior to arming, a warning will be spawned advising the user to RTL. This message is meant to be issued while flying. A quick investigation showed that the condition is broken.

**Describe your solution**
Fix the condition and use an armed boolean instead of the landed flag.

### CODE REVIEW

**Comment by dagar** [comment]
Obviously the RTL part is wrong/misleading when disarmed, but are we sure this should all be silenced until armed? What about flagging actual failures like if an airspeed (differential pressure) sensors stops responding while sitting around before flight?

**Comment by fury1895** [comment]
@dagar I asked myself the same question. Initially I wanted to add a message for the case you are mentioning. But then I thought, how likely is it that your airspeed is flagged valid and afterwards invalid before takeoff? For these IMO rather rare cases, is it not enough to spawn the airspeed invalid message when trying to arm?

On the other hand, adding an additional message that is spawned "live" is not a big deal so why not. 
@sfuhrer what would be your 2 cents on that topic?

File: src/modules/airspeed_selector/airspeed_selector_main.cpp [hunk 1]
```diff
@@ -150,6 +150,7 @@ class AirspeedModule : public ModuleBase<AirspeedModule>, public ModuleParams,
 	float _ground_minus_wind_TAS{0.0f}; /**< true airspeed from groundspeed minus windspeed */
 	float _ground_minus_wind_CAS{0.0f}; /**< calibrated airspeed from groundspeed minus windspeed */
 	bool _armed_prev{false};
+	bool _armed{false};
```
Comment (sfuhrer) [inline src/modules/airspeed_selector/airspeed_selector_main.cpp:153]: The same information as in _armed can be read with `_vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED `correct?
I wouldn't make a global variable out of it. That adds an extra state, and the more states the more complexity often.

**Review by sfuhrer — COMMENTED** [review]
Yes I agree that the notification about a sensor getting invalid should also appear in the disarmed state, with a different wording than when armed.

File: src/modules/airspeed_selector/airspeed_selector_main.cpp (hunk 1 above)
Comment (fury1895) [reply src/modules/airspeed_selector/airspeed_selector_main.cpp]: resolved with [6d9d261](https://github.com/PX4/PX4-Autopilot/pull/19173/commits/6d9d261e78a0ebc0019ee05f3bc2320c302fb4b1)

**Comment by fury1895** [comment]
@sfuhrer @dagar please have a look at the latest commit. I am not sure if it is the best way to setup the condition

File: src/modules/airspeed_selector/airspeed_selector_main.cpp [hunk 2]
```diff
@@ -593,9 +593,14 @@ void AirspeedModule::select_airspeed_and_publish()
 
 	// print warning or info, depending of whether airspeed got declared invalid or healthy
 	if (_valid_airspeed_index != _prev_airspeed_index &&
-	    (_number_of_airspeed_sensors > 0 || !_vehicle_land_detected.landed) &&
-	    _valid_airspeed_index != _prev_airspeed_index) {
-		if (_prev_airspeed_index > 0) {
+	    _number_of_airspeed_sensors > 0) {
+		if (_vehicle_status.arming_state == !vehicle_status_s::ARMING_STATE_ARMED && _prev_airspeed_index > 0) {
+			mavlink_log_critical(&_mavlink_log_pub,
+					     "Airspeed sensor failure detected. Check connection between sensor and flight controller.\t");
```
Comment (sfuhrer) [inline src/modules/airspeed_selector/airspeed_selector_main.cpp:599]:
```suggestion
					     "Airspeed sensor failure detected. Check connection and reboot\t");
```
By default the airspeed validator doesn't declare the sensor valid again if the data was once dropped, so if the user will want to arm a reboot first is required.

Comment (fury1895) [reply src/modules/airspeed_selector/airspeed_selector_main.cpp]: addressed with the last commit, thx for having a look!

**Review by sfuhrer — APPROVED** [review]

**Review by sfuhrer — CHANGES_REQUESTED** [review]
image
CI isn't happy @fury1895

**Pull request merged**